src/test/java/org/jenkinsci/plugins/durabletask/EncodingTest.java

```diff
@@ -67,23 +68,21 @@
 @RunWith(Parameterized.class)
 public class EncodingTest {
 
-    @ClassRule public static JenkinsRule r = new JenkinsRule();
+    @Rule public LoggerRule logging = new LoggerRule().recordPackage(BourneShellScript.class, Level.FINE);
+    @Rule public JenkinsRule r = new JenkinsRule();
+    @Rule public DockerClassRule<JavaContainer> dockerUbuntu = new DockerClassRule<>(JavaContainer.class);
```

Comment:
I think it would also work if you kept everything as `@ClassRule` and `static`, but combined `unixAndDocker` and `setUp`, which might make the tests a little quicker. My best guess as to the problem is that `setUp` was actually running first (AIUI there is no guaranteed ordering between multiple `@Before(Class)` methods), when we want `unixAndDocker` to run first.

Reply:
That's actually what I tried as well, but then I was still failing. It wasn't until I realized there was an '@AfterClass' method that I managed to resolve the issues.  I'll try going back to `@ClassRule` and tweaking the `@AfterClass`